### PR TITLE
[search] Update list of features skipped from search

### PR DESCRIPTION
**Pull request by pastk**
Following https://github.com/organicmaps/organicmaps/pull/9851 and https://github.com/organicmaps/organicmaps/pull/9854

### CODE REVIEW

File: search/types_skipper.cpp [hunk 1]
```diff
@@ -44,6 +52,7 @@ void TypesSkipper::SkipEmptyNameTypes(feature::TypesHolder & types) const
     if (m_isPoi(type))
       return false;
 
+    ftype::TruncValue(type, 2);
```
Comment (vng) [inline search/types_skipper.cpp:55]: The point here is about an exact match, and the comment before was:
```
Test for exact type (man_made-tower-communication is not).
```
Had a request to search for cellular towers :)

File: indexer/ftypes_matcher.cpp [hunk 2]
```diff
@@ -461,8 +461,11 @@ TwoLevelPOIChecker::TwoLevelPOIChecker() : ftypes::BaseChecker(2 /* level */)
       {"man_made", "water_tap"},
       {"man_made", "water_well"},
       {"natural", "beach"},
-      {"natural", "geyser"},
       {"natural", "cave_entrance"},
+      {"natural", "geyser"},
+      {"natural", "hot_spring"},
+      {"natural", "peak"},
+      {"natural", "saddle"},
```
Comment (vng) [inline indexer/ftypes_matcher.cpp:468]: I'd prefer not to treat peak, saddle as POIs.

Comment (pastk) [reply indexer/ftypes_matcher.cpp:468]: What does it affect besides the search?

Comment (vng) [reply indexer/ftypes_matcher.cpp:468]: IsPoiChecker

File: search/types_skipper.cpp (hunk 1 above)
Comment (pastk) [reply search/types_skipper.cpp:55]: There are many other tower types that could be unnamed (minor) but worth to search for, e.g. bell towers, observation towers, fort towers, pagodas
See https://wiki.openstreetmap.org/wiki/Key:tower:type

Comment (pastk) [reply search/types_skipper.cpp:55]: So probably its best to index all of them until we separate them into more subtypes.

File: indexer/ftypes_matcher.cpp (hunk 2 above)
Comment (pastk) [reply indexer/ftypes_matcher.cpp:468]: Yes, but what is it used for? :)

File: search/types_skipper.cpp (hunk 1 above)
Comment (vng) [reply search/types_skipper.cpp:55]: We have only one 3-arity type now - communication.

File: indexer/ftypes_matcher.cpp (hunk 2 above)
Comment (vng) [reply indexer/ftypes_matcher.cpp:468]: Search ranking, Editor, Wiki builder, some lists, 6 occurencies.

File: search/types_skipper.cpp (hunk 1 above)
Comment (pastk) [reply search/types_skipper.cpp:55]: Yes, this is what I mean.
There are many search cases now under a `man_made=tower`.

(Which should ideally have their own types like a cell tower.)

File: indexer/ftypes_matcher.cpp (hunk 2 above)
Comment (pastk) [reply indexer/ftypes_matcher.cpp:475]: I've checked all usages I could find (see my new comments) and I see no issue with adding peaks/saddles.

**Comment by pastk** [comment]
- make `man_made-tower` searchable (many cases in https://wiki.openstreetmap.org/wiki/Key:tower:type)
- map `man_made=mast tower:type=communication` into `man_made-tower-communication`
- add comments

**Comment by vng** [comment]
So:
- I see issues adding peaks and saddles as POIs and I disagree with it
- tower-communications are not searchable with these changes

Instead of fixing a bad merge from the previous PR (that I didn't notice) we introduce a bunch of new controversial changes.

I propose to make international changes, but not let's now "fix everything" ..

**Comment by pastk** [comment]
>     * I see issues adding peaks and saddles as POIs and I disagree with it

These are important POIs for hikers.
Could you please elaborate your concern?

>     * Instead of tower-communications, now we gonna index all towers (what for?)

E.g. a user had been told about an observation tower in the area. How to find it?
There are many other [tower types](https://wiki.openstreetmap.org/wiki/Key:tower:type) a user might want to search for (and they could be unnamed).

**Comment by vng** [comment]
Generally, my concerns are the same as I told before in search-category PRs.
I'd like to see the unblocking fix now, and then make a separate PR with towers and peaks to test.

**Comment by vng** [comment]
Remove "ftype::TruncValue(type, 2);" to keep _communication_ search as before

**Comment by pastk** [comment]
> Remove "ftype::TruncValue(type, 2);" to keep _communication_ search as before

It'll break the test, I see no reason to revert it back and then revert again..

**Comment by pastk** [comment]
> Generally, my concerns are the same as I told before in search-category PRs. I'd like to see the unblocking fix now, and then make a separate PR with towers and peaks to test.

- https://github.com/organicmaps/organicmaps/pull/9895

**Pull request merged**